eZ/Publish/Core/Persistence/Cache/UserHandler.php

```diff
@@ -288,9 +288,9 @@ public function loadRoleByIdentifier($identifier, $status = Role::STATUS_DEFINED
         }
 
         return $this->getCacheValue(
-            $identifier,
+            $this->escapeForCacheKey($identifier),
             'ez-role-',
-            function ($identifier) {
+            function () use ($identifier) {
```

Comment:
Is this change necessary?

Reply:
without this change, we will use the variable after using the escapeForCacheKey function, and we need the original value to search in the database